### DISCUSSION TITLE
InfoWindow headerContent renders html as string

### DISCUSSION THREAD

**Discussion by mikechalmers** [inocan-group/vue3-google-map#298] [Q&A] [answered]
When using InfoWindow, rendering html as content seems work just fine, but the headerContent - which also allows html - renders the same html as a string:

```
const InfoWindowOptions = {
  content: '<h1>InfoWindow</h1>',
  headerContent: '<h1>Header</h1>',
};
```

<img width="236" alt="Screenshot 2024-11-30 at 23 07 18" src="https://github.com/user-attachments/assets/a9315451-89fa-4623-ab3e-de641bfac02f">

**Comment by HusamElbashir** [comment, marked as answer]
Looks like there is a slight difference in the way `content` and `headerContent` behave. If you refer to the [docs](https://developers.google.com/maps/documentation/javascript/reference/info-window#InfoWindowOptions):
 - `content`: Content to display in the InfoWindow. This can be an HTML element, a plain-text string, or **a string containing HTML**.
 - `headerContent`: The content to display in the InfoWindow header row. This can be an HTML element, or a string of plain text.

The difference being that `content` can take a string containing HTML but `headerContent` cannot. You're going to have to use an actual HTML element here:

```
const header = document.createElement('h1')
header.textContent = 'Header'

const InfoWindowOptions = {
  headerContent: header,
};
```

**Comment by mikechalmers** [comment]
You're totally right, that worked like a charm. Thank you for your speedy, helpful answer.
